discovery-provider/plugins/network-monitoring/src/metrics/index.ts

```diff
@@ -1,102 +1,121 @@
+  const partiallySyncedUserCount = await getPartiallySyncedUsersCount(run_id);
 
-    const unsyncedUsersCount = await getUnsyncedUsersCount(run_id)
+  const unsyncedUsersCount = await getUnsyncedUsersCount(run_id);
 
-    const usersWithNullPrimaryClock = await getUsersWithNullPrimaryClock(run_id)
+  const usersWithNullPrimaryClock = await getUsersWithNullPrimaryClock(run_id);
 
-    const usersWithAllFoundationNodeReplicaSetCount = await getUsersWithEntireReplicaSetInSpidSetCount(run_id, foundationNodes)
+  const usersWithAllFoundationNodeReplicaSetCount =
+    await getUsersWithEntireReplicaSetInSpidSetCount(run_id, foundationNodes);
 
-    allUserCount.forEach(({ endpoint, count }) => {
-        allUserCountGauge.set({ endpoint, run_id }, count)
-    })
-    primaryUserCount.forEach(({ endpoint, count }) => {
-        primaryUserCountGauge.set({ endpoint, run_id }, count)
-    })
+  allUserCount.forEach(({ endpoint, count }) => {
+    allUserCountGauge.set({ endpoint, run_id }, count);
+  });
+  primaryUserCount.forEach(({ endpoint, count }) => {
+    primaryUserCountGauge.set({ endpoint, run_id }, count);
+  });
 
-    fullySyncedUsersCountGauge.set({ run_id }, fullySyncedUsersCount)
-    partiallySyncedUsersCountGauge.set({ run_id }, partiallySyncedUserCount)
-    unsyncedUsersCountGauge.set({ run_id }, unsyncedUsersCount)
-    nullPrimaryUsersCountGauge.set({ run_id }, usersWithNullPrimaryClock)
-    usersWithAllFoundationNodeReplicaSetGauge.set({ run_id }, usersWithAllFoundationNodeReplicaSetCount)
+  userCountGauge.set({ run_id }, userCount);
+  fullySyncedUsersCountGauge.set({ run_id }, fullySyncedUsersCount);
+  partiallySyncedUsersCountGauge.set({ run_id }, partiallySyncedUserCount);
+  unsyncedUsersCountGauge.set({ run_id }, unsyncedUsersCount);
+  nullPrimaryUsersCountGauge.set({ run_id }, usersWithNullPrimaryClock);
+  usersWithAllFoundationNodeReplicaSetGauge.set(
+    { run_id },
+    usersWithAllFoundationNodeReplicaSetCount
+  );
 
-    // Record duration for generating metrics and export to prometheus
-    endTimer({ run_id: run_id })
+  // Record duration for generating metrics and export to prometheus
+  endTimer({ run_id: run_id });
 
+  if (userCount > 0) {
     await publishSlackReport({
-        fullySyncedUsersCount: fullySyncedUsersCount,
-        partiallySyncedUsersCount: partiallySyncedUserCount,
-        unsyncedUsersCount: unsyncedUsersCount,
-        usersWithNullPrimaryClock: usersWithNullPrimaryClock
-    })
-
-    try {
-        // Finish by publishing metrics to prometheus push gateway
-        console.log(`[${run_id}] pushing metrics to gateway`);
-        await gateway.pushAdd({ jobName: 'network-monitoring' })
-    } catch (e) {
-        console.log(`[generateMetrics] error pushing metrics to pushgateway - ${(e as Error).message}`)
-    }
-
-
-    console.log(`[${run_id}] finish generating metrics`);
-}
+      fullySyncedUsersCount:
+        ((fullySyncedUsersCount / userCount) * 100).toFixed(2) + "%",
```

Comment:
might be valuable to show absolute number and percent here but nbd

Reply:
For slack, a percentage is probably good tbh. Grafana will always have the number

Reply:
agreed